Make armv7-linux-androideabi default target on Android

MortimerGoro opened the pull request:
<!-- Please describe your changes on the following line: -->

See https://github.com/servo/servo/issues/11921

After fixing problematic dependencies this is the last step to support armv7-linux-androideabi target on Android:

- Updates skia dependency to fix a linker error. See https://github.com/servo/skia/pull/136
- Fixes specifying android targets on `./mach package` and `./mach install` steps:
  -`./mach package --release --target=arm-linux-androideabi`
  -`./mach package --release --target=armv7-linux-androideabi`
  -`./mach package --release --target=aarch64-linux-android`
- Make `armv7-linux-androideabi`default when `--android` param is used in build, package or install commands

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #11921 (github issue number if applicable).

<!-- Either: -->
- [x] There are tests for these changes OR
- [ ] These changes do not require tests because _____

<!-- Also, please make sure that "Allow edits from maintainers" checkbox is checked, so that we can help you if you get stuck somewhere along the way.-->

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/17008)
<!-- Reviewable:end -->

MortimerGoro committed:
Make armv7-linux-androideabi default target on Android

highfive commented:
Heads up! This PR modifies the following files:
 * @wafflespeanut: python/servo/package_commands.py, python/servo/command_base.py

emilio commented:
what about CI? Does it use the `--android` flag?

r? @larsbergstrom

MortimerGoro commented:
> what about CI? Does it use the --android flag?

Yes, it uses the --android flag, which will translate to armv7-linux-androideabi after this PR is merged.

I also plan to add add aarch64-linux-android target compilation to the CI when all the dependencies are ready (waiting for https://github.com/carllerche/mio/pull/607 be ready in crates.io)

bors-servo commented:
:umbrella: The latest upstream changes (presumably #17042) made this pull request unmergeable. Please resolve the merge conflicts.

jdm commented:
@bors-servo: r=larsberg

bors-servo commented:
:pushpin: Commit fdbf68e has been approved by `larsberg`

<!-- @bors-servo r=larsberg fdbf68e662e9331b703abdc8fb9919a35bf6392c -->

bors-servo commented:
:hourglass: Testing commit fdbf68e662e9331b703abdc8fb9919a35bf6392c with merge b0313163e4b600c7ee9db68764cd69a53c17eaa3...

bors-servo commented:
:broken_heart: Test failed - [android](http://build.servo.org/builders/android/builds/7254)

MortimerGoro commented:
Fixed the hardcoded path in one of the a ci steps.
```
objdump_output = subprocess.check_output([
    os.path.join(
        os.environ['ANDROID_NDK'], 'toolchains', 'arm-linux-androideabi-4.9',
        'prebuilt', 'linux-x86_64', 'bin', 'arm-linux-androideabi-objdump'),
    '-T',
    'target/arm-linux-androideabi/debug/libservo.so']
).split(b'\n')
```

jdm commented:
@bors-servo r=larsberg

bors-servo commented:
:pushpin: Commit 04fb628 has been approved by `larsberg`

<!-- @bors-servo r=larsberg 04fb628d20168b903f3554dbf8dc394e217b6119 -->

bors-servo commented:
:hourglass: Testing commit 04fb628d20168b903f3554dbf8dc394e217b6119 with merge a694f70acf2f011da816a24716d6501ab153c64e...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/7257), [arm32](http://build.servo.org/builders/arm32/builds/7473), [arm64](http://build.servo.org/builders/arm64/builds/7509), [linux-dev](http://build.servo.org/builders/linux-dev/builds/7598), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/4356), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/4376), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/7581), [mac-rel-css1](http://build.servo.org/builders/mac-rel-css1/builds/65), [mac-rel-css2](http://build.servo.org/builders/mac-rel-css2/builds/64), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/4283), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/4348), [mac-rel-wpt3](http://build.servo.org/builders/mac-rel-wpt3/builds/66), [mac-rel-wpt4](http://build.servo.org/builders/mac-rel-wpt4/builds/67), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/2825)
Approved by: larsberg
Pushing a694f70acf2f011da816a24716d6501ab153c64e to master...
